pennylane/operation.py

```diff
@@ -582,21 +582,79 @@ def decompose(self):
             return self.decomposition(wires=self.wires)
         return self.decomposition(*self.parameters, wires=self.wires)
 
+    @staticmethod
+    def compute_diagonalizing_gates(*params, wires, **hyperparams):
+        r"""Defines a partial representation of this operator via
+        its eigendecomposition.
+
+        Given the eigendecomposition `O = U \Sigma U^{\dagger}` where
+        :math:`\Sigma` is a diagonal matrix containing the eigenvalues,
+        the sequence of diagonalizing gates implements the unitary :math:`U`.
+        In other words, the diagonalizing gates rotate the state into the eigenbasis
+        of this operator.
+
+        This is the static version of ``diagonalizing_gates``, which can be called
+        without creating an instance of the class.
+
+        .. note::
+
+            This method gets overwritten by subclasses to define the representation of a particular operator.
+            By default, this method should always take the operator's parameters, wires and hyperparameters as
+            inputs (even if the diagonalizing gates are independent of these values).
+
+            Alternatively, a custom signature can be defined, in which case the `diagonalizing_gates()`
+            method has to be overwritten to use the right signature.
```

Comment:
I hope this communicates the idea to developers?

Reply:
But it may be important if you're trying to do overwrite the method? So you remember that you have to overwrite something else too?

Reply:
Yeah - I guess my direction of thought was more: if I was a user reading this docstring (because I want to use `op.diagonalizing_gates()`, I would be confused by this statement, and might think I have to subclass or something.

So I like delineating quite explicitly "For developers or people that are making their own operations:..."